app/ModelFunctions/PhotoFunctions.php

```diff
@@ -169,33 +172,24 @@ public function add(array $file, $albumID_in = 0, $delete_imported = false, $for
 			return Response::error('An upload-folder is missing or not readable and writable!');
 		}
 
+		$public = 0;
+		$star = 0;
+		$albumID = null;
+
 		switch ($albumID_in) {
-				// s for public (share)
-			case 's':
+			case 'public':
 				$public = 1;
-				$star = 0;
-				$albumID = null;
 				break;
 
-				// f for starred (fav)
-			case 'f':
+			case 'starred':
 				$star = 1;
-				$public = 0;
-				$albumID = null;
 				break;
 
-				// r for recent
-				// 0 for unsorted
-			case '0':
-			case 'r':
-				$public = 0;
-				$star = 0;
-				$albumID = null;
+			case 'unsorted':
+			case '0': // root
```

Comment:
Does this handle (or need to handle) integer 0?

Reply:
nope does not need to. Inputs are seen as string.